app/controllers/partners/family_requests_controller.rb

```diff
@@ -11,23 +11,23 @@ def new
         params[:filterrific]
       ) || return
       @children = @filterrific.find
+
+      @children_items = flash[:children_items] || {}
+      @error_messages = flash[:error_messages]
     end
 
     def create
-      children_ids = []
+      children_items = params[:children_items] || {} # Assuming this is how you structured the data
+      family_requests_attributes = []
 
-      params.each do |key, _|
-        is_child, id = key.split('-')
-        if is_child == 'child'
-          children_ids << id
-        end
-      end
+      children_items.each do |child_id, item_ids|
+        child = current_partner.children.active.find_by(id: child_id)
+        next unless child
 
-      children = current_partner.children.active.where(id: children_ids).where.not(item_needed_diaperid: [nil, 0])
-
-      children_grouped_by_item_id = children.group_by(&:item_needed_diaperid)
-      family_requests_attributes = children_grouped_by_item_id.map do |item_id, item_requested_children|
-        { item_id: item_id, person_count: item_requested_children.size, children: item_requested_children }
+        item_ids.each do |item_id|
+          # Assuming each item can be requested once per child in a request
+          family_requests_attributes << { item_id: item_id, person_count: 1, child: child }
```

Comment:
Do Items have any permissions to go with them?

If not, this is fine. If they do, this pattern would allow a user to arbitrarily specific any item_id and potentially bypass permissions.